packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -23,19 +23,25 @@ class TimelineData {
   //  and frame ids are available in the engine.
   final int cpuThreadId;
   final int gpuThreadId;
-  String frameId;
-
-  // Maps frame ids to their respective frames.
-  final Map<String, TimelineFrame> _frames = <String, TimelineFrame>{};
-
-  final List<TimelineEvent> _pendingEvents = [];
 
   final StreamController<TimelineFrame> _frameCompleteController =
       StreamController<TimelineFrame>.broadcast();
   Stream<TimelineFrame> get onFrameCompleted => _frameCompleteController.stream;
 
-  final Map<String, TimelineEvent> _asyncEvents = <String, TimelineEvent>{};
+  /// Frames we are in the process of forming.
+  ///
+  /// Once frames are ready, we will remove them from this Map and add them to
+  /// [_frameCompleteController].
+  final Map<String, TimelineFrame> _pendingFrames = <String, TimelineFrame>{};
+
+  /// Events we have collected and are waiting to add to their respective
+  /// frames.
+  final List<TimelineEvent> _pendingEvents = [];
+
+  /// The current stack of duration events for CPU work.
   TimelineEvent _cpuDurationStack;
```

Comment:
how is a single frame a stack? I would have expected a stack would be a 
List<TimelineEvent> or similar.

Reply:
Each event is really a tree - has a single parent and a list of children. I just left original naming, but perhaps it would be better as cpuEventTree and gpuEventTree. Wdyt?

Reply:
Changed to cpuEventNode and gpuEventNode